lib/state_machine.js

```diff
@@ -76,6 +79,12 @@ limitations under the License.
        */
       this.context_ = undefined;
     
+      /**
+       * An error listener, which is invoked when an error occurs.
+       * The default one just swallows all errors.
```

Comment:
Can it not console.log them or something?

Reply:
We do that whenever an exception is generated (see the place where this listener is invoked). We assume that folks always want to see the error and that we always want to go back to the initial state in the SM. This allows folks to perform custom behavior on top of that (for example, alerting a higher-level SM that we need to transition to another module).